OsmAnd-java/build.gradle

```diff
@@ -124,6 +124,11 @@ publishing {
 }
 
 dependencies {
+	if (project.findProject(':OsmAnd') != null) {
```

Comment:
You can specify parameters via settings.gradle
For android ```shared_config=jvmPublicConfig``` -- https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties
```
implementation project(path:':OsmAnd-shared', configuration:<Project.property.shared_config>)
```